docs/src/dev/understanding_traces.md

```diff
@@ -0,0 +1,39 @@
+# Understanding Traces
+
+yk can print the traces it has created to `stdout` to help with debugging.
+However, these traces are often lengthy, and not always easy to understand.
+This section briefly explains how to get yk to print its traces, and how
+to make them a bit easier to understand.
+
+
+## Producing a trace
+
+The `YKD_PRINT_IR` environment variable determines whether yk prints traces to
+stdout or not. If `jit-pre-opt` is specified, the traces will not be optimised,
```

Comment:
This sounds like the trace compiler doesn't do any optimisation if you print the pre-opt IR.